src/lib/reasoners/fun_sat.ml

```diff
@@ -1621,16 +1621,24 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
         );
     Util.loop
       ~f:(fun _n () acc ->
-          let new_guard = E.fresh_name Ty.Tbool in
-          save_guard_and_refs acc new_guard;
-          let guards = ME.add new_guard
-              (mk_gf new_guard "" true true,Ex.empty)
-              acc.guards.guards in
-          {acc with guards =
-                      { acc.guards with
-                        current_guard = new_guard;
-                        guards = guards;
-                      }})
+          try
+            let new_guard = E.fresh_name Ty.Tbool in
+            save_guard_and_refs acc new_guard;
+            let guards = ME.add new_guard
+                (mk_gf new_guard "" true true,Ex.empty)
+                acc.guards.guards in
+            {acc with guards =
+                        { acc.guards with
+                          current_guard = new_guard;
+                          guards = guards;
+                        }}
+          with
+          | Util.Step_limit_reached _ ->
+            (* If we reached the step limit while pushing, we have to cancel the
+               push.
+               TODO: properly cancel the push. *)
+            env
```

Comment:
Can you explain why the steps limit increases while pushing? We call: `E.fresh_name`, `save_guard_and_refs` (which only pushes things on a stack), `mk_gf` (which is just a dumb record constructor) and `ME.add`, none of which seem to increase the step count.

Reply:
I thought `fun_sat` and `satml_frontend` has the same behavior regarding the steps; I was wrong.